test/metabase/cmd/dump_to_h2_test.clj

```diff
@@ -67,43 +66,43 @@
 (deftest dump-to-h2-dump-plaintext-test
   (testing "dump-to-h2 --dump-plaintext"
     (let [h2-fixture-db-file  @cmd.test-util/fixture-db-file-path
-          h2-file-plaintext   (format "/tmp/out-%s.db" (mt/random-name))
-          h2-file-enc         (format "/tmp/out-%s.db" (mt/random-name))
-          h2-file-default-enc (format "/tmp/out-%s.db" (mt/random-name))
           db-name             (str "test_" (mt/random-name))]
-      (mt/test-drivers #{:h2 :postgres :mysql}
-        (with-redefs [i18n.impl/site-locale-from-setting-fn (atom (constantly false))]
-          (binding [setting/*disable-cache*    true
-                    mdb.connection/*db-type*   driver/*driver*
-                    mdb.connection/*jdbc-spec* (persistent-jdbcspec driver/*driver* db-name)
-                    db/*db-connection*         (persistent-jdbcspec driver/*driver* db-name)
-                    db/*quoting-style*         driver/*driver*]
-            (when-not (= driver/*driver* :h2)
-              (tx/create-db! driver/*driver* {:database-name db-name}))
-            (load-from-h2/load-from-h2! h2-fixture-db-file)
-            (eu/with-secret-key "89ulvIGoiYw6mNELuOoEZphQafnF/zYe+3vT+v70D1A="
-              (db/insert! Setting {:key "my-site-admin", :value "baz"})
-              (db/update! Database 1 {:details "{\"db\":\"/tmp/test.db\"}"})
-              (dump-to-h2/dump-to-h2! h2-file-plaintext {:dump-plaintext? true})
-              (dump-to-h2/dump-to-h2! h2-file-enc {:dump-plaintext? false})
-              (dump-to-h2/dump-to-h2! h2-file-default-enc))
+      (mt/with-temp-file [h2-file-plaintext (format "/tmp/out-%s.db" (mt/random-name))]
```

Comment:
should there be a `mt/with-temp-files` to unnest all of this?

Reply:
I *was* going to make `with-temp-file` allow multiple bindings, but since I the 3rd arg is optional file contents I thought it wouldn't be clear how a multiple binding version should look, or potentially confusing if there are different semantics between two different versions. Since this is the only place in the entire codebase I could find that created multiple temp files at once I decided against it.

Since n oone actually uses the `[absolute-filename filename-and-extension contents]` version at this point in time maybe it makes sense to get rid of that version and then `with-temp-file` can allow multiple bindings

Reply:
good point about this being the only place. no need to complicate this if its one call site.

Reply:
On second thought, after typing that all out I decided it would be more useful to support multiple bindings after all than to support inline file contents. I tweaked the test utils to support that